plugin.video.edltv/default.py

```diff
@@ -0,0 +1,317 @@
+# -*- coding: utf-8 -*-
+# Module: default
+# Author: Roman V. M.
+# Created on: 28.11.2014
+# License: GPL v.3 https://www.gnu.org/copyleft/gpl.html
+import sqlite3
+import json
+import sys
+import os
+import socket
+import filecmp
+from urllib import urlencode
+from urlparse import parse_qsl
+import xbmcgui
+import xbmcplugin
+import xbmcvfs
+
+_url = sys.argv[0]
+_handle = int(sys.argv[1])
+string = ""
+word_list = []
+
+def check_internet_connected () :
+    for timeout in [1,5,10,15]:
+        try:
+            print "checking internet connection.."
+            socket.setdefaulttimeout(timeout)
+            host = socket.gethostbyname("www.google.com")
+            s = socket.create_connection((host, 80), 2)
+            s.close()
+            return True
+        except Exception,e:
+            print e
+    return False
+
+def set_period(code) :
+    rows = c.execute('SELECT * FROM edltv WHERE parent LIKE ?',(code,))
+    for row in rows :
+        period.append(row[0])
+    if period :
+        maxperiod = str(max(period))[7:8]
+        if maxperiod == "0":
+            list_videos(str(max(period))[0:8])
+        else :
+            for i in range(1, int(maxperiod)+1) :
+                data = 'คาบที่ '+str(i)
+                period_code = code+str(i)
+                list_item = xbmcgui.ListItem(label=data)
+                url = get_url(action='listing', category=period_code, hasfolder='no')
+                is_folder = True
+                xbmcplugin.addDirectoryItem(_handle, url, list_item, is_folder)
+            xbmcplugin.endOfDirectory(_handle)
+    else :
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok('eDLTV', 'ขออภัย หัวข้อนี้ไม่มีวิดีโอ')
```

Comment:
This is not a blocker for me personally (not sure what others think) but I'd rather prefer you to use translations instead of hardcoding strings. see: http://kodi.wiki/view/Language_support